resolwe_bio/processes/reads_processing/trimmomatic.yml

```diff
@@ -131,65 +151,111 @@
-      {% endif %}
-
-      echo "Removing adapters and filtering data..."
-
-      {% if adapters %}
-        echo "Preparing adapter file..."
-        ADAPTERS=`basename {{adapters.fasta.file}} .fasta.gz`
-        gzip -cd {{ adapters.fasta.file }} > "${ADAPTERS}.fasta"
-        re-progress 0.1
-      {% endif %}
-
       NAME=`basename {{reads.fastq.0.file}} .fastq.gz`
 
       READS=({% for read in reads.fastq %}{{read.file}} {% endfor %})
       cat "${READS[@]}" > in_reads.fastq.gz
       re-progress 0.2
 
-      TrimmomaticSE \
-        -threads {{threads}} \
-        -trimlog \
-        stats.txt \
-        in_reads.fastq.gz \
-        out_reads_filtered.fastq.gz \
-        {% if adapters %}"ILLUMINACLIP:${ADAPTERS}.fasta:"{{ illuminaclip.seed_mismatches }}":30:"{{ illuminaclip.simple_clip_threshold }}{% endif %} \
-        {% if slidingwindow.window_size and slidingwindow.required_quality %}"SLIDINGWINDOW:"{{ slidingwindow.window_size }}":"{{ slidingwindow.required_quality }}{% endif %} \
-        {% if maxinfo.target_length and maxinfo.strictness %}}"MAXINFO:"{{maxinfo.target_length}}":"{{maxinfo.strictness}}{% endif %} \
-        {% if leading %}"LEADING:"{{ leading }}{% endif %} \
-        {% if trailing %}"TRAILING:"{{ trailing }}{% endif %} \
-        {% if crop %}"CROP:"{{ crop }}{% endif %} \
-        {% if headcrop %}"HEADCROP:"{{ headcrop }}{% endif %} \
-        {% if minlen %}"MINLEN:"{{minlen}}{% endif %} \
-        {% if average_quality %}"AVGQUAL:"{{ average_quality }}{% endif %}
-
-      re-checkrc "Failed while processing with Trimmomatic."
-      re-progress 0.7
-
-      mv out_reads_filtered.fastq.gz "${NAME}_filtered.fastq.gz"
-
-      echo "Postprocessing FastQC..."
-
-      # http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
-      mkdir "fastqc" && fastqc "${NAME}_filtered.fastq.gz" --extract --outdir="fastqc" >&2
-      re-checkrc "Failed while processing with FastQC."
-      re-progress 0.9
-
-      mv "fastqc/${NAME}_filtered_fastqc" "fastqc/${NAME}_fastqc"
-      mv "fastqc/${NAME}_filtered_fastqc.zip" "${NAME}_fastqc.zip"
-
-      FASTQC_URL="{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}"
-
-      re-progress 0.95
-
-      re-save-file-list fastq "${NAME}_filtered.fastq.gz"
-      re-save-file stats stats.txt
+      {% if
+        not reads_filtering.average_quality
+        and not reads_filtering.minlen
+        and not trim_bases.headcrop
+        and not trim_bases.crop
+        and not trim_bases.trailing
+        and not trim_bases.leading
+        and not maxinfo.target_length
+        and not  maxinfo.strictness
+        and not slidingwindow.window_size
+        and not slidingwindow.required_quality
+        and not illuminaclip.adapters
+        and not illuminaclip.seed_mismatches
+        and not illuminaclip.simple_clip_threshold %}
+        re-warning "To process reads with Trimmomatic please define at least one parameter
```

Comment:
I have changed to: ""The reads are to be copied from the input to the output, including the quality control report. To process reads with Trimmomatic please define at least one parameter." I don't know about the length of Error message, I thought this will be seen only in standard output of the process. Are these Error messages shown also somewhere else on frontend?